backend/operations_scanner.go

```diff
@@ -270,30 +350,43 @@ func (s *OperationsScanner) updateOperationStatus(ctx context.Context, logger *s
 	if err != nil {
 		return err
 	}
-	if updated {
-		logger.Info(fmt.Sprintf("Updated Resources container item for '%s' with provisioning state '%s'", doc.ExternalID, opStatus))
-	}
 
 	return nil
 }
 
-func (s *OperationsScanner) maybePostAsyncNotification(ctx context.Context, logger *slog.Logger, doc *database.OperationDocument) {
-	if len(doc.NotificationURI) > 0 {
-		err := s.postAsyncNotification(ctx, doc.NotificationURI)
+// maybePostAsyncNotification attempts to notify ARM of a completed asynchronous
+// operation if the initial request included an "Azure-AsyncNotificationUri" header.
+func (s *OperationsScanner) maybePostAsyncNotification(ctx context.Context, op operation) {
+	if len(op.doc.NotificationURI) > 0 {
+		err := s.postAsyncNotification(ctx, op.id)
 		if err == nil {
-			logger.Info(fmt.Sprintf("Posted async notification for operation '%s'", doc.ID))
+			op.logger.Info("Posted async notification")
 		} else {
-			logger.Error(fmt.Sprintf("Failed to post async notification for operation '%s': %s", doc.ID, err.Error()))
+			op.logger.Error(fmt.Sprintf("Failed to post async notification: %v", err.Error()))
 		}
 	}
 }
 
-func (s *OperationsScanner) postAsyncNotification(ctx context.Context, url string) error {
-	request, err := http.NewRequestWithContext(ctx, http.MethodPost, url, nil)
+// postAsyncNotification submits an POST request with status payload to the given URL.
+func (s *OperationsScanner) postAsyncNotification(ctx context.Context, operationID string) error {
+	// Refetch the operation document to provide the latest status.
+	doc, err := s.dbClient.GetOperationDoc(ctx, operationID)
+	if err != nil {
+		return err
+	}
```

Comment:
In case you're wondering why this refetch is necessary, it's the same reason as in
https://github.com/Azure/ARO-HCP/pull/985#issuecomment-2546136032